File: LeetCode/Medium/Problems 151 - 200/Problem 167/p167Sol2.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* twoSum(int* numbers, int numbersSize, int target, int* returnSize){
    int i = 0;
    int j = numbersSize - 1;
    
    while (i < j) {
        int twoSum = numbers[i] + numbers[j];
        if (twoSum == target) {
            *returnSize = 2;
            int* sol = malloc(2 * sizeof(int));
            sol[0] = i + 1; 
            sol[1] = j + 1;
            return sol;
        }
        else if (twoSum < target) {
            i++;
        }
        else {
            j--;
        }
    }
    
    return 0;
}

File: LeetCode/Medium/Problems 151 - 200/Problem 167/p167Sol2.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        i = 0
        j = len(numbers) - 1
        
        while i < j:
            twoSum = numbers[i] + numbers[j]
            if twoSum == target:
                return [i+1, j+1]
            elif twoSum < target:
                i += 1
            else:
                j -= 1